deploy/monitor/push_service_monitor.yaml

```diff
@@ -7,8 +7,8 @@ metadata:
   name: unifiedpush
 spec:
   endpoints:
-    - path: /api/metrics
-      port: server
+    - path: /rest/prometheus/metrics
+      port: web
   selector:
     matchLabels:
-      app: unifiedpush
\ No newline at end of file
+      service: example-unifiedpushserver-unifiedpush
```

Comment:
actually, let's change this back to the old one `app: unifiedpush` as it doesn't change so we can reuse this file in the installation.

Reply:
Hmm, let's see. We provision 2 Services for UPS, one for the Route (goes through the oauth-proxy), and one for internal access (originally from within the cluster, like this, I think). For either Service, the `app` and `service` labels are generated based on the CR name (both change, not ideal), so are not really appropriate to statically use here.

Assuming we want to only collect metrics through the internal Service (the one that doesn't use the oauth-proxy), then we could set the selector here to `internal: unifiedpush`: https://github.com/aerogear/unifiedpush-operator/blob/master/pkg/controller/unifiedpushserver/unifiedpushserver.go#L341

Reply:
yeah, good idea.

Reply:
Done 👍

Reply:
why do you think the `internal: unifiedpush` label selector will not work?

Reply:
BTW, this has been applied here https://master.demo27-8353.openshiftworkshop.com/console/project/mobile-unifiedpush/edit/yaml?kind=ServiceMonitor&name=unifiedpush&group=monitoring.coreos.com&returnURL=project%2Fmobile-unifiedpush%2Fbrowse%2Fother%3Fkind%3DServiceMonitor%26group%3Dmonitoring.coreos.com and it works fine.

Reply:
It is done as you requested and deployed in the env as described in the first comment.

Reply:
Without the service worked 👍